### ISSUE TITLE
API: BatchNorm1d.parameters should respect affine=False

### ISSUE THREAD

**Issue by lewisnjue** [lewisnjue/nnetflow#31]
layers.BatchNorm1d currently returns [gamma, beta] from parameters() regardless of affine flag. When affine=False, these are frozen (requires_grad=False) but are still returned as parameters, which is surprising for optimizers.\n\nProposed fix:\n- Have parameters() return [] when affine=False.\n- Optionally, only create gamma/beta tensors when affine=True.\n\nAdd a test to confirm no trainable params are returned in the non-affine case.\n
